moveit_ros/moveit_servo/include/moveit_servo/servo_parameters.cpp

```diff
@@ -0,0 +1,278 @@
+                                      node, logger);
+  error |= !declareAndGetParam<double>(ns + ".joint_limit_margin", parameters->joint_limit_margin, node, logger);
+
+  // Collision checking
+  error |= !declareAndGetParam<bool>(ns + ".check_collisions", parameters->check_collisions, node, logger);
+  error |= !declareAndGetParam<double>(ns + ".collision_check_rate", parameters->collision_check_rate, node, logger);
+  error |= !declareAndGetParam<std::string>(ns + ".collision_check_type", parameters->collision_check_type, node, logger);
+  error |= !declareAndGetParam<double>(ns + ".self_collision_proximity_threshold",
+                                      parameters->self_collision_proximity_threshold, node, logger);
+  error |= !declareAndGetParam<double>(ns + ".scene_collision_proximity_threshold",
+                                      parameters->scene_collision_proximity_threshold, node, logger);
+  error |= !declareAndGetParam<double>(ns + ".collision_distance_safety_factor", parameters->collision_distance_safety_factor,
+                                      node, logger);
+  error |= !declareAndGetParam<double>(ns + ".min_allowable_collision_distance", parameters->min_allowable_collision_distance,
+                                      node, logger);
+
+  // Only continue if all parameters were found
+  if (error)
+  {
+    RCLCPP_ERROR_STREAM(logger, "One or more Servo parameters missing, check YAML file before proceeding");
+    return false;
+  }
+
+  // Begin input checking
+  if (parameters->publish_period <= 0.)
+  {
+    RCLCPP_WARN(logger, "Parameter 'publish_period' should be "
+                        "greater than zero. Check yaml file.");
+    return false;
+  }
+  if (parameters->num_outgoing_halt_msgs_to_publish < 0)
+  {
+    RCLCPP_WARN(logger, "Parameter 'num_outgoing_halt_msgs_to_publish' should be greater than zero. Check yaml file.");
+    return false;
+  }
+  if (parameters->hard_stop_singularity_threshold < parameters->lower_singularity_threshold)
+  {
+    RCLCPP_WARN(logger, "Parameter 'hard_stop_singularity_threshold' "
+                        "should be greater than 'lower_singularity_threshold.' "
+                        "Check yaml file.");
+    return false;
+  }
+  if ((parameters->hard_stop_singularity_threshold < 0.) || (parameters->lower_singularity_threshold < 0.))
+  {
+    RCLCPP_WARN(logger, "Parameters 'hard_stop_singularity_threshold' "
+                        "and 'lower_singularity_threshold' should be "
+                        "greater than zero. Check yaml file.");
+    return false;
+  }
+  if (parameters->low_pass_filter_coeff < 0.)
+  {
+    RCLCPP_WARN(logger, "Parameter 'low_pass_filter_coeff' should be "
+                        "greater than zero. Check yaml file.");
+    return false;
+  }
+  if (parameters->joint_limit_margin < 0.)
+  {
+    RCLCPP_WARN(logger, "Parameter 'joint_limit_margin' should be "
```

Comment:
.. should be greater than or equal to zero

Reply:
Should this also be changed in ROS1? [See here](https://github.com/ros-planning/moveit/blob/589f83a132da174991b233931d3ca7f31ab1cba4/moveit_ros/moveit_servo/src/servo.cpp#L196-L201)

Reply:
IMO that's not significant enough to merit a PR in ROS1, but might as well fix it here.

(Feel free to make a ROS1 PR if you want to, but we already have like 2-3 open PRs that need reviewing)

Reply:
Done